docs/rfcs/0005-json-library.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: page
+title: RFC process
+permalink: /rfc/0005-json-library
+---
+
+* Start date: 2020-07-28
+* RFC PR: TBD
+
+## Summary
+
+Introduce a library for reading and writing JSON into Android Components and Fenix.
+
+## Motivation
+
+We currently interact with JSON in many areas of Android Components and Fenix. Currently serialization and deserialization is all done manually with `JSONObject`, which creates more work for developers and potential bugs. Fenix also is interested in using a library and we'd like to avoid shipping multiple JSON libraries to users. Using a JSON library allows us to reduce code that needs to be written, potentially gain some performance wins from a well maintained lib, and avoid bugs in our custom JSON implementations.
+
+Some examples of JSON usage today are:
+- Reading a `WebAppManifest`
+- Parsing API calls in various service components
+- Sending messages to and from extensions
+
+## Guide-level explanation
+
+We'll start with selecting a JSON library to integrate into Android Components and Fenix. Some options we've looked at include:
+- [kotlinx-serialization](https://github.com/Kotlin/kotlinx.serialization): Multi-platform, multi-format, built by JetBrains. Also much newer than other solutions so documentation is relatively lacking.
+- [Gson](https://github.com/google/gson): Long standing library built by Google. Slower than other options.
+- [Moshi](https://github.com/square/moshi): Simpler API, built by Square. Fewer features than Gson but also faster.
```

Comment:
I think using Gson or Moshi (as much as I like them) is kind of a violation of our design axioms: We'd essentially start packaging those libraries into every app that uses AC.
`kotlinx.serialization` feels more in line, as an extension of the Kotlin standard library, that we explicitly make use of. The last time I looked at it, it was full of "experimental" annotations and didn't seem ready for production use yet though. Of course the APK size impact is still important for this library.

Reply:
It seems that the code is stable and production ready, but the API surface may change between versions.